test/meta.cpp

```diff
@@ -81,378 +84,1988 @@ void test_tuple_cat()
+                "");
+            static_assert(std::is_same<reverse<list<int, short, double, float>>,
+                                       list<float, double, short, int>>::value,
+                          "");
+            static_assert(
+                std::is_same<reverse<list<int[1], int[2], int[3], int[4], int[5], int[6], int[7],
+                                          int[8], int[9], int[10], int[11], int[12], int[13],
+                                          int[14], int[15], int[16], int[17]>>,
+                             list<int[17], int[16], int[15], int[14], int[13], int[12], int[11],
+                                  int[10], int[9], int[8], int[7], int[6], int[5], int[4], int[3],
+                                  int[2], int[1]>>::value,
+                "");
+
+            /**
+             * \sa `meta::pop_front`
+             */
+            static_assert(
+                std::is_same<pop_front<list<int, short, int, double, short, double, double>>,
+                             list<short, int, double, short, double, double>>::value,
+                "");
+
+            /**
+             * \sa `meta::push_back`
+             */
+            static_assert(
+                std::is_same<push_back<list<int, short, int, double, short, double, double>, float>,
+                             list<int, short, int, double, short, double, double, float>>::value,
+                "");
+
+            /**
+             * \sa `meta::push_front
+             */
+            static_assert(
+                std::is_same<
+                    push_front<list<int, short, int, double, short, double, double>, float>,
+                    list<float, int, short, int, double, short, double, double>>::value,
+                "");
+
+            /**
+             * \sa `meta::sort`
+             */
+
+            using unsorted_list =
+                list<char[5], char[3], char[2], char[6], char[1], char[5], char[10]>;
+            using sorted_list =
+                meta::sort<unsorted_list,
+                           lambda<_a, _b, lazy::less<lazy::sizeof_<_a>, lazy::sizeof_<_b>>>>;
+            using expected_list =
+                list<char[1], char[2], char[3], char[5], char[5], char[6], char[10]>;
+            static_assert(std::is_same<sorted_list, expected_list>::value, "");
+
+            /**
+             * \sa `meta::partition`
+             */
+            using unpartitioned_list =
+                list<char[5], char[3], char[2], char[6], char[1], char[5], char[10]>;
+
+            namespace detail
+            {
+                struct is_even
+                {
+                    template <typename N>
+                    using invoke = bool_<N::type::value % 2 == 0>;
+                };
+                static_assert(!meta::invoke<not_fn<quote<is_even::invoke>>, int_<2>>::value, "");
+
+                template <typename L>
+                using
```

Comment:
This struct (preexisting) is unused and can be deleted.

Reply:
struct removed, merge conflict resolved. Committed/built.